### PR TITLE
Update RUNSQLFLAGS to recursive assignment from simple to allow for o…

### PR DESCRIPTION
**Pull request by irfanshar**
…verridden TGTRLS

### CODE REVIEW

File: src/mk/def_rules.mk [hunk 1]
```diff
@@ -387,6 +387,10 @@ SQLCBLIPGM_OBJTYPE := *PGM
 SQLCBLIPGM_OPTION := $(OPTION)
 SQLCBLIPGM_TGTRLS := $(TGTRLS)
 
+SQLPRC_TGTRLS := $(TGTRLS)
+
+SQLTRG_TGTRLS := $(TGTRLS)
+
```
Comment (edmundreinhardt) [inline src/mk/def_rules.mk:393]: so why specifically for SQLPRC and SQLTRG, what about all the others.  Why not have a generic SQL_TGTRLS?
Does the customer use TGTRLS or SQLPRC_TGTRLS in their Rules.mk?

Comment (irfanshar) [reply src/mk/def_rules.mk:393]: https://github.com/IBM/ibmi-bob/blob/fd101c793e391c212eca0f926fa017e14c804a63/src/mk/def_rules.mk#L388
https://github.com/IBM/ibmi-bob/blob/fd101c793e391c212eca0f926fa017e14c804a63/src/mk/def_rules.mk#L382

If you notice, we previously have specific TGTRLS for our different types of SQL. This change was made to follow our previous format.

The customer did not specify anything to do with TGTRLS, but COMMIT.

However, to fix the bug we had switched to recursive assignment, which exposed that these 2 types were being passed in `UNKNOWN_FILE_TYPE` due to the resolution of `programTGTRLS`

Comment (edmundreinhardt) [reply src/mk/def_rules.mk:393]: As you can see from here https://ibm.github.io/ibmi-bob/#/welcome/features?id=support-sql-pseudo-source
there are a few more extensions that we should also support such as SQALIAS, SQLSEQ, SQLUDF, SQLUDT

File: src/mk/def_rules.mk [hunk 2]
```diff
@@ -765,14 +771,14 @@ moduleTGTRLS = $(strip \
...
 	$(if $(filter %.cblle,$<),$(CBLMOD_TGTRLS), \
-	$(if $(filter %.SQLC,$<),$(SQLCIMOD_TGTRLS), \
-	$(if $(filter %.sqlc,$<),$(SQLCIMOD_TGTRLS), \
-	$(if $(filter %.SQLCPP,$<),$(SQLCPPIMOD_TGTRLS), \
-	$(if $(filter %.sqlcpp,$<),$(SQLCPPMOD_TGTRLS), \
-	$(if $(filter %.SQLRPGLE,$<),$(SQLRPGIMOD_TGTRLS), \
-	$(if $(filter %.sqlrpgle,$<),$(SQLRPGIMOD_TGTRLS), \
-	$(if $(filter %.SQLCBLLE,$<),$(SQLCBLIMOD_TGTRLS), \
-	$(if $(filter %.sqlcblle,$<),$(SQLCBLIMOD_TGTRLS), \
+	$(if $(filter %.SQLC,$<),$(SQL_TGTRLS), \
```
Comment (edmundreinhardt) [inline src/mk/def_rules.mk:774]: I would recommend CMOD_TGTRLS 
This should be tied more to the compiler and not to RUNSQLMOD

File: src/mk/def_rules.mk [hunk 3]
```diff
@@ -765,14 +771,14 @@ moduleTGTRLS = $(strip \
...
+	$(if $(filter %.sqlc,$<),$(SQL_TGTRLS), \
+	$(if $(filter %.SQLCPP,$<),$(SQL_TGTRLS), \
```
Comment (edmundreinhardt) [inline src/mk/def_rules.mk:776]: I would recommned CPPMOD_TGTRLS

File: src/mk/def_rules.mk [hunk 4]
```diff
@@ -765,14 +771,14 @@ moduleTGTRLS = $(strip \
...
+	$(if $(filter %.sqlcpp,$<),$(SQL_TGTRLS), \
+	$(if $(filter %.SQLRPGLE,$<),$(SQL_TGTRLS), \
```
Comment (edmundreinhardt) [inline src/mk/def_rules.mk:778]: RPGMOD_TGTRLS

File: src/mk/def_rules.mk [hunk 5]
```diff
@@ -765,14 +771,14 @@ moduleTGTRLS = $(strip \
...
+	$(if $(filter %.sqlrpgle,$<),$(SQL_TGTRLS), \
+	$(if $(filter %.SQLCBLLE,$<),$(SQL_TGTRLS), \
```
Comment (edmundreinhardt) [inline src/mk/def_rules.mk:780]: CBLMOD_TGTRLS

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]

**Review by edmundreinhardt — APPROVED** [review]
Good!

**Pull request merged**